irohad/main/iroha_conf_loader.cpp

```diff
@@ -0,0 +1,300 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "main/iroha_conf_loader.hpp"
+
+#include <fstream>
+#include <limits>
+#include <sstream>
+
+#include <rapidjson/document.h>
+#include <rapidjson/error/en.h>
+#include <rapidjson/istreamwrapper.h>
+#include <rapidjson/rapidjson.h>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/range/adaptor/map.hpp>
+#include "main/iroha_conf_literals.hpp"
+
+#include "main/assert_config.hpp"
+
+namespace ac = assert_config;
+namespace mbr = config_members;
+
+static constexpr size_t kBadJsonPrintLength = 15;
+static constexpr size_t kBadJsonPrintOffsset = 5;
+static_assert(kBadJsonPrintOffsset <= kBadJsonPrintLength,
+              "The place of error is out of the printed string boundaries!");
+
+/**
+ * Adds the children logger configs from parent logger JSON object to parent
+ * logger config. The parent logger JSON object is searched for the children
+ * config section, and the children configs are parsed and created if the
+ * section is present.
+ * @param parent_config - the parent logger config
+ * @param parent_obj - the parent logger json configuration
+ */
+void addChildrenLoggerConfigs(
+    const std::string &path,
+    const logger::LoggerManagerTreePtr &parent_config,
+    const rapidjson::Value::ConstObject &parent_obj);
+
+/**
+ * Overrides the logger configuration with the values from JSON object.
+ * @param dst - the configuration to use as base
+ * @param src - the JSON object to take overrides from
+ */
+void updateLoggerConfig(const std::string &path,
+                        logger::LoggerConfig &cfg,
+                        const rapidjson::Value::ConstObject &obj);
+
+/**
+ * Gets a value by a key from a JSON object, if present.
+ * @param dest - the variable to store the value
+ * @param obj - the source JSON object
+ * @param key - the key for the requested value
+ * @return true if the value was loaded, otherwise false.
+ */
+template <typename TDest, typename TKey>
+bool tryGetValByKey(const std::string &path,
+                    TDest &dest,
```

Comment:
I think, template parameters here and in other functions in this file should be accepted by a universal reference, not a lvalue one